src/inmanta/agent/agent.py

```diff
@@ -311,43 +311,50 @@ async def execute(
         self, dummy: "ResourceActionBase", generation: "Dict[ResourceIdStr, ResourceActionBase]", cache: AgentCache
     ) -> None:
         await dummy.future
-        try:
-            result = await self.scheduler.get_client().get_resource(
-                self.scheduler.agent._env_id,
-                self.resource_id.resource_version_str(),
-                logs=True,
-                log_action=const.ResourceAction.deploy,
-                log_limit=1,
-            )
-            if result.code != 200 or result.result is None:
-                LOGGER.error("Failed to get the status for remote resource %s (%s)", str(self.resource_id), result.result)
-                return
+        async with self.scheduler.agent.process.cad_ratelimiter:
+            try:
+                # got event or cancel first
+                if self.is_done():
+                    return
```

Comment:
Should we log something here? The other `execute` implementations do.

Reply:
This one is kind of an invisible internal mechanism.

This is not ideal, as this is becoming slow and slow/invisible is a bad combination.